### PR TITLE
Move OnMeshStart call below clear mesh update calls to ease handling cycling through index in implementations

### PR DESCRIPTION
**Pull request by InsanityAutomation**
Adds additional callbacks for Homing, Stepper State, and Mesh started. Follows methods used by @sebazzz and brings them to the ext_ui.h header instead of the machine specific files, adds default functions to other affected screens.

Edit : the remaining change relocates the OnMeshStart call below where the mesh is zeroed out so that the second duplicated mesh handler originally added to avoid duplicating calls that indexed a counter both on clear then again on write is no longer needed.

### CODE REVIEW

**Comment by sjasonsmith** [comment]
It seems the merges eliminated all but this last change. Can you update the title/description accordingly?

I can probably surmise the reason for the change, but I'd like to be able to fact-check it against your description.

File: Marlin/src/gcode/bedlevel/abl/G29.cpp [hunk 1]
```diff
@@ -399,6 +397,10 @@ G29_TYPE GcodeSuite::G29() {
       points[0].z = points[1].z = points[2].z = 0;  // Probe at 3 arbitrary points
     #endif
 
+    #if BOTH(AUTO_BED_LEVELING_BILINEAR, EXTENSIBLE_UI)
+      ExtUI::onMeshLevelingStart();
+    #endif
```
Comment (sjasonsmith) [inline Marlin/src/gcode/bedlevel/abl/G29.cpp:402]: Is this only used for Bilinear? Not MBL or UBL? Previously it looks like it was always enabled, regardless of leveling.

Comment (InsanityAutomation) [reply Marlin/src/gcode/bedlevel/abl/G29.cpp:402]: MBL and UBL should have it in their respective files, this file only handles 3 point, linear, and bilinear.

Comment (sjasonsmith) [reply Marlin/src/gcode/bedlevel/abl/G29.cpp:402]: Oh, of course. I didn't look at which file I was in.

Comment (Sebazzz) [reply Marlin/src/gcode/bedlevel/abl/G29.cpp:402]: > 
> 
> Is this only used for Bilinear? Not MBL or UBL? Previously it looks like it was always enabled, regardless of leveling.

Shouldn't that statement apply that to the PR of strain gauge support as well?

Comment (InsanityAutomation) [reply Marlin/src/gcode/bedlevel/abl/G29.cpp:402]: Thats in the other PR, and everything for that is implemented directly in the probe class so it should not care what leveling routine is calling it.

**Pull request merged**